var/spack/repos/builtin/packages/esmf/package.py

```diff
@@ -430,5 +471,10 @@ def check(self):
     def setup_dependent_build_environment(self, env, dependent_spec):
         env.set("ESMFMKFILE", os.path.join(self.prefix.lib, "esmf.mk"))
 
-    def setup_run_environment(self, env):
-        env.set("ESMFMKFILE", os.path.join(self.prefix.lib, "esmf.mk"))
+    def install(self, pkg, spec, prefix):
+        make("install")
+
+        if "+python" in spec:
+            # build the python library
+            python_builder = PythonPipBuilder(pkg)
```

Comment:
(no change needed) This is an interesting strategy I haven't seen elsewhere: 

* Inheriting `PythonPackage` instead of `PythonExtension` would cause problems because that introduces another build system (which would conflict with the one introduced by inheriting `MakefilePackage`)
* But in your case you always use the makefile build and then conditionally run the pip build I haven't seen any other packages that access the builder in this way (i.e. explicitly, vs. setting it up to run implicitly via inheritance)

Thanks to @climbfuji and @Chrismarsh for verifying this runs

Reply:
Haha glad you like it. Took me a while to wrap my head around. Although there are a couple other packages that use a similar pattern which is where I got the idea.